src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -420,42 +438,32 @@ public static GapicProductConfig createDummyInstance(
         continue;
       }
       interfaceConfigProtos.put(interfaceConfigProto.getName(), interfaceConfigProto);
+      protoInterfaces.put(interfaceConfigProto.getName(), apiInterface);
     }
 
-    // Parse proto file for interfaces.
-    for (ProtoFile file : sourceProtos) {
-      if (file.getProto().getServiceList().size() == 0) continue;
-      for (DescriptorProtos.ServiceDescriptorProto service : file.getProto().getServiceList()) {
-        String serviceFullName =
-            String.format("%s.%s", file.getProto().getPackage(), service.getName());
-        Interface apiInterface = symbolTable.lookupInterface(serviceFullName);
-        if (apiInterface == null) {
-          diagCollector.addDiag(
-              Diag.error(SimpleLocation.TOPLEVEL, "interface not found: %s", service.getName()));
-          continue;
-        }
-        InterfaceConfigProto interfaceConfigProto = interfaceConfigProtos.get(serviceFullName);
-        if (interfaceConfigProto == null) {
-          interfaceConfigProto = InterfaceConfigProto.getDefaultInstance();
-        }
-        String interfaceNameOverride = languageSettings.getInterfaceNamesMap().get(serviceFullName);
-
-        GapicInterfaceConfig interfaceConfig =
-            GapicInterfaceConfig.createInterfaceConfig(
-                diagCollector,
-                language,
-                defaultPackageName,
-                interfaceConfigProto,
-                apiInterface,
-                interfaceNameOverride,
-                messageConfigs,
-                resourceNameConfigs,
-                protoParser);
-        if (interfaceConfig == null) {
-          continue;
-        }
-        interfaceConfigMap.put(serviceFullName, interfaceConfig);
+    // Generate GapicInterfaceConfigs from the proto interfaces and interfaceConfigProtos.
+    for (String serviceFullName : protoInterfaces.keySet()) {
+      InterfaceConfigProto interfaceConfigProto =
+          interfaceConfigProtos.getOrDefault(
+              serviceFullName, InterfaceConfigProto.getDefaultInstance());
+      Interface apiInterface = protoInterfaces.get(serviceFullName);
```

Comment:
I think it could be better to iterate `protoInterfaces.entrySet()`

Reply:
oh yah then we don't have to do the lookup good call